### PR TITLE
feat: multicompany

### PR DESCRIPTION
**Pull request by FH-sandeepkaur1**
## Change Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/entry/list-organizations/list-organizations.component.ts [hunk 1]
```diff
@@ -7,9 +10,21 @@ import { Component, OnInit } from '@angular/core';
...
 
-  constructor() { }
+  currentUser:any = {};
+  organizations = [];
+  constructor(private apiService: ApiService, private router: Router,) { }
 
-  ngOnInit(): void {
+  async ngOnInit() {
+    this.currentUser = (await Auth.currentSession()).getIdToken().payload;
+    console.log('currentUser', this.currentUser)
```
Comment (kunalfleethawks) [inline src/app/entry/list-organizations/list-organizations.component.ts]: remove console log

File: src/app/entry/list-organizations/list-organizations.component.ts [hunk 2]
```diff
@@ -7,9 +10,21 @@ import { Component, OnInit } from '@angular/core';
...
+    this.apiService.getData(`carriers/get/organizations/${this.currentUser.carrierID}`).subscribe((res) => {
+      this.organizations = res;
+    })
   }
 
+  switchDashboard(data) {
+    localStorage.setItem('xfhCarrierId', data.carrierID);
```
Comment (kunalfleethawks) [inline src/app/entry/list-organizations/list-organizations.component.ts:25]: we need to secure this as if someone tampers it - they can manipulate carriers.

File: src/app/pages/accounts/sales/sales-orders/add-sales-order/add-sales-order.component.ts [hunk 3]
```diff
@@ -220,8 +220,10 @@ export class AddSalesOrderComponent implements OnInit {
   }
 
   getCurrentuser = async () => {
-    this.currentUser = (await Auth.currentSession()).getIdToken().payload;
-    this.salesData.salePerson = `${this.currentUser.firstName} ${this.currentUser.lastName}`;
+    // this.currentUser = (await Auth.currentSession()).getIdToken().payload;
```
Comment (kunalfleethawks) [inline src/app/pages/accounts/sales/sales-orders/add-sales-order/add-sales-order.component.ts:223]: Remove commented code

File: src/app/pages/dispatch/new-documents/my-documents/my-document-list.component.ts [hunk 4]
```diff
@@ -123,8 +123,9 @@ export class MyDocumentListComponent implements OnInit {
   }
 
   getCurrentuser = async () => {
-    this.currentUser = (await Auth.currentSession()).getIdToken().payload;
-    this.currentUser = `${this.currentUser.firstName} ${this.currentUser.lastName}`;
+    // this.currentUser = (await Auth.currentSession()).getIdToken().payload;
+    // this.currentUser = `${this.currentUser.firstName} ${this.currentUser.lastName}`;
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/new-documents/my-documents/my-document-list.component.ts:127]: remove all commented code

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
Please have look at minor comments

File: src/app/entry/list-organizations/list-organizations.component.ts (hunk 2 above)
Comment (FH-sandeepkaur1) [reply src/app/entry/list-organizations/list-organizations.component.ts:25]: Can you please the method for the same

**Review by kunalfleethawks — APPROVED** [review]

**Pull request merged**